### PR TITLE
diff-so-fancy 0.3.0 (new formula)

### PR DESCRIPTION
**Pull request by OJFord**
Closes stevemao/diff-so-fancy#21.

### CODE REVIEW

File: Library/Formula/diff-so-fancy.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class DiffSoFancy < Formula
+  desc "builds on the good-lookin' output of diff-highlight"
+  homepage "https://github.com/stevemao/diff-so-fancy"
+  url "https://github.com/stevemao/diff-so-fancy/archive/v0.1.2.tar.gz"
+  sha256 "fa4ac51368ba6b942c6b346117e5f72cabb00ea3c29f834e3c4ca80972e19502"
+
+  depends_on "gnu-sed"
```
Comment (DomT4) [inline Library/Formula/diff-so-fancy.rb]: Run time or just build time?

Comment (DomT4) [inline Library/Formula/diff-so-fancy.rb]: Ah, run time, looking at the `inreplace` below.

File: Library/Formula/diff-so-fancy.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def install
+    prefix.install Dir["*"]
+    inreplace "#{prefix}/diff-so-fancy", "sed", "gsed"
```
Comment (DomT4) [inline Library/Formula/diff-so-fancy.rb]: It'll be a little cleaner to do this before you install it, and we'd prefer it to be tucked in `libexec` and symlinked from there ideally:
```ruby
# Short explanation like: requires functionality not offered by BSD sed.
inreplace "diff-so-fancy", "sed", "gsed"
libexec.install Dir["*"]
bin.install_symlink libexec/"diff-so-fancy"
```

File: Library/Formula/diff-so-fancy.rb (hunk 1 above)
Comment (OJFord) [inline Library/Formula/diff-so-fancy.rb]: Yep, run time. The `inreplace` avoids need for default names install flag, and avoids issues if already installed without it.

File: Library/Formula/diff-so-fancy.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    bin.install_symlink "#{prefix}/diff-so-fancy"
+  end
+
+  def caveats; <<-EOS.undent
```
Comment (DomT4) [inline Library/Formula/diff-so-fancy.rb]: Unless these are Homebrew-specific we prefer not to include them. People should reference upstream for general usage.

File: Library/Formula/diff-so-fancy.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    You may also want some fancier colours. Consider trying:
+      git config --global color.diff-highlight.oldNormal "red bold"
+      git config --global color.diff-highlight.oldHighlight "red bold 52"
+      git config --global color.diff-highlight.newNormal "green bold"
+      git config --global color.diff-highlight.newHighlight "green bold 22"
+    EOS
+  end
+
+  test do
+    system "diff-so-fancy"
```
Comment (DomT4) [inline Library/Formula/diff-so-fancy.rb]: ` system bin/"diff-so-fancy"`

**Comment by OJFord** [comment]
CI fail reason, not clear from the log, is that the repo is not 30+ days old.

With 1,630 stars and 24 forks though it's hopefully clearly not going anywhere.

File: Library/Formula/diff-so-fancy.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/diff-so-fancy.rb]: Would be good to make a feature request to be able to use BSD sed.

Comment (OJFord) [inline Library/Formula/diff-so-fancy.rb]: @mikemcquaid Not likely to be worked on/accepted, the author (@paulirish) says in stevemao/diff-so-fancy#25 "OS X sed is crap".

Comment (mikemcquaid) [inline Library/Formula/diff-so-fancy.rb]: Disappointing but: thanks.

File: Library/Formula/diff-so-fancy.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+class DiffSoFancy < Formula
+  desc "builds on the good-lookin' output of diff-highlight"
```
Comment (stevemao) [inline Library/Formula/diff-so-fancy.rb]: I changed the desc to "Good-lookin' diffs with diff-highlight and more" on npm and "Unfancy git-diff" on GitHub

File: Library/Formula/diff-so-fancy.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "https://github.com/stevemao/diff-so-fancy"
+  url "https://github.com/stevemao/diff-so-fancy/archive/v0.1.2.tar.gz"
+  sha256 "fa4ac51368ba6b942c6b346117e5f72cabb00ea3c29f834e3c4ca80972e19502"
+
+  depends_on "gnu-sed"
+
+  def install
+    inreplace "diff-so-fancy", "sed", "gsed"
```
Comment (stevemao) [inline Library/Formula/diff-so-fancy.rb]: gsed and sed should be both supported. gsed takes higher precedence if both available.

Comment (mikemcquaid) [inline Library/Formula/diff-so-fancy.rb]: In that case: let's just use `sed`.

Comment (mikemcquaid) [inline Library/Formula/diff-so-fancy.rb]: Er, let's avoid the replace, I mean.

Comment (stevemao) [inline Library/Formula/diff-so-fancy.rb]: That should work.

Comment (OJFord) [inline Library/Formula/diff-so-fancy.rb]: @stevemao So why does [the readme say](https://github.com/stevemao/diff-so-fancy/blob/master/readme.md#install-gnu-sed):

> Install GNU sed
> You'll need GNU sed. On Mac, it's easy to install with Homebrew.
> ```brew install gnu-sed```

**Comment by paulirish** [comment]
Aye. We don't need the inreplace.  Thanks for already taking care of that, @OJFord 

We'll work to remove the dependency on gnu-sed, in the short term.

File: Library/Formula/diff-so-fancy.rb (hunk 6 above)
Comment (stevemao) [inline Library/Formula/diff-so-fancy.rb]: Yes, Thanks for taking care of this @OJFord :)

**Comment by DomT4** [comment]
Does this need updating to `v0.2.0` now?

**Comment by OJFord** [comment]
@DomT4 Hadn't seen that, will update now.

**Comment by stevemao** [comment]
@DomT4 yes, but by waiting for another 25+ days we will be sure to make several releases

**Comment by DomT4** [comment]
To be honest, unless any other maintainers object, I'm fairly content to waive the 30 day requirement here. There's enough Issues and PRs being created by people who aren't upstream developers to convince me there's significant community interest in the project.

**Comment by stevemao** [comment]
Oh well, that's great :) thanks :)

**Comment by OJFord** [comment]
Should we wait for stevemao/diff-so-fancy#55 though?

Would remove the `gnu-sed` dependancy if it's correct, but I'm not familiar enough with GNU vs BSD sed to know how complete the PR is :smile:

**Comment by DomT4** [comment]
Well, there's no massive rush. I'll at least give fellow maintainers a chance to comment on the 30 day issue; they object when I act like a Dictator :stuck_out_tongue_winking_eye:.

**Comment by stevemao** [comment]
Hi guys, just letting you know that we have moved the project to an org. 
https://github.com/so-fancy/diff-so-fancy
Sorry for the trouble.

**Comment by DomT4** [comment]
Looks like `v0.3.0` has been cut now. Is that all we're waiting on or are there additional issues to resolve?

**Comment by stevemao** [comment]
> diff-so-fancy fails to resolve the location of the included diff-highlight script as of v2.1; may as well install it anyway.

@OJFord If there is a problem, [report it](https://github.com/so-fancy/diff-so-fancy/issues/new) 

@DomT4 Any released version could be published here.

**Comment by OJFord** [comment]
@stevemao I don't think it's a bug your end, it's just something changed that meant it didn't work with the symlink I was using before. I didn't dig deeper because this seemed a better solution anyway.

@DomT4 As far as I can tell from the changelog, it's just administrative stuff - CI, readme - updating  anyway though.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/2084df34970fc0afe3312673f824f763f75ae6c5. Thank you for your contribution to Homebrew @OJFord; we appreciate it! :smiley_cat:

**Pull request closed without merging**